Bring in color system

gabalafou opened the pull request:
Closes #14482.

This pull request takes a minimal approach, clamping existing colors to similar colors in the palette so long as the colors seemed close enough. In other words, this pull request avoids fundamental changes of hue - for example, this pull request does no change the focus ring color from gray to mauve.

- [x] issues: #14482
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

gabalafou committed:
Bring in color system

gabalafou commented:
I updated all of the baseline snapshot images so reviewers can see how this pull request changes the visual appearance of the UI.

gabalafou committed:
update link

gabalafou commented on bokehjs/test/baselines/linux/Bug__in_issue_#12447__make_tooltips_interfere_with_toolbars.png:
Hmm, this snapshot may indicate a problem. The hover tooltip notch/wedge/data pointer is more visible on the left than on the right. Will investigate.

gabalafou commented on bokehjs/test/baselines/linux/Bug__in_issue_#12585__doesn't_allow_support_for_line_policy=none_with_mode=vline.png:
But here the darker tooltip notch looks good, being against a lighter background

gabalafou commented on bokehjs/test/baselines/linux/Icons__should_support_all_icons_defined_in_less_icons.less.png:
This is odd. Only some of these icons look like they are darker now

mosc9575 replied:
What you see is the visual effect of the open task https://github.com/bokeh/bokeh/issues/12382. Not all icons are of type `svg` yet. Some are still `png` and loaded from `docs/bokeh/source/_images/icons`.

gabalafou replied:
Yep, after rebasing, this looks fixed now

gabalafou commented on bokehjs/test/baselines/linux/Plot__should_allow_toolbar_placement_below_inside_the_frame.png:
Weird, I didn't know you could put the toolbar inside the plot

gabalafou commented on bokehjs/test/baselines/linux/Tools__should_support_BoxSelectTool's_setup_menu.png:
I think this matches the snapshot of all the icons in that this set does not look darker to my eyes.

gabalafou replied:
After rebasing, it's darker now.

gabalafou reviewed (commented):
I reviewed all of the snapshot images, left a few remarks. Would be good if somebody else can look at them.

gabalafou committed:
Update baselines

gabalafou commented:
@mattpap, I rebased latest from `branch-3.8` and updated the snapshot baselines. Could you give this a look and/or let me know what next steps are needed?

mattpap committed:
Improve contrast of Tooltip's arrow in any conditions

mattpap commented:
Improved contrast of `Tooltip`'s arrow.

* on dark background (after/before):
<img width="650" height="250" alt="image" src="https://github.com/user-attachments/assets/a6ab2b76-7f65-481d-934f-b756c79807bf" />
<img width="650" height="250" alt="image" src="https://github.com/user-attachments/assets/3819ea4c-34f9-4f61-bc06-5abd8081c54a" />

* on light background (after/before):
<img width="250" height="250" alt="image" src="https://github.com/user-attachments/assets/da0a7a96-e472-4d0d-8088-4f563e168e70" />
<img width="250" height="250" alt="image" src="https://github.com/user-attachments/assets/f08f6c34-50ac-4666-a06f-6372130787f2" />

mattpap committed:
Update baseline images
